lib/components/helpers/array-item-control.js

```diff
@@ -44,12 +44,16 @@ export default React.createClass({
       readOnly: isLastItem && !config.isRemovalOfLastArrayItemAllowed(field)
     });
 
-    const moveItemForward = this.props.index < (this.props.numItems - 1) ?
-      config.createElement('move-item-forward', { field: field, onClick: this.onMoveForward }) :
+    const moveItemForward = field.fieldIndex !== field.parent.value.length - 1 ?
```

Comment:
Shouldn't `this.props.numItems` be the same as `field.parent.value.length`? I'd like to keep the parent responsible for that, rather than this control having to peek at the parent value.

I'm not quite following how this changes the logic from what it was before: only showing the forward button if it's not the last item.